isis/src/base/objs/StatsFunc/StatsFunc.cpp

```diff
@@ -0,0 +1,158 @@
+#include "StatsFunc.h"
+
+#include <string>
+#include <iostream>
+
+#include "Application.h"
+#include "CubeAttribute.h"
+#include "Cube.h"
+#include "FileName.h"
+#include "Histogram.h"
+#include "Process.h"
+#include "Pvl.h"
+#include "UserInterface.h"
+
+using namespace std;
+using namespace Isis;
+
+namespace Isis {
+  void stats(std::vector<char*>& args) {
+    // TODO is this line bad?
+    UserInterface ui("/work/users/jmapel/ISIS3/isis/src/base/apps/stats/stats.xml", args);
+
+    Cube *inputCube = new Cube();
+    CubeAttributeInput inAtt(ui.GetAsString("FROM"));
+    inputCube->setVirtualBands(inAtt.bands());
+    inputCube->open(ui.GetFileName("FROM"));
+
+    double validMin = Isis::ValidMinimum;
+    double validMax = Isis::ValidMaximum;
+
+    if ( ui.WasEntered("VALIDMIN") ) {
+      validMin = ui.GetDouble("VALIDMIN");
+    }
+
+    if ( ui.WasEntered("VALIDMAX") ) {
+      validMax = ui.GetDouble("VALIDMAX");
+    }
+
+    Pvl statsPvl = stats(inputCube, validMin, validMax);
+
+    for (int resultIndex = 0; resultIndex < statsPvl.groups(); resultIndex++) {
+      if (statsPvl.group(resultIndex).name() == "Results") {
+        Application::Log(statsPvl.group(resultIndex));
```

Comment:
Create a dummy application object at the top of this function to make it so it doesn't segfault if run outside of an app